SPECS/tdnf/virtual-repo-snapshot.patch

```diff
@@ -0,0 +1,900 @@
+From 9d4b634ca7be64c4167c9f4361498e5fe42e664d Mon Sep 17 00:00:00 2001
+From: Sam Meluch <sammeluch@microsoft.com>
+Date: Tue, 30 Apr 2024 13:56:44 -0700
+Subject: [PATCH] Add virtual repo snapshot feature to tdnf
+
+---
+ client/config.c                            |   5 +
+ client/prototypes.h                        |   3 +-
+ client/repo.c                              |  50 +-
+ common/config.h                            |   1 +
+ etc/bash_completion.d/tdnf-completion.bash |   2 +-
+ include/tdnferror.h                        |   9 +
+ include/tdnftypes.h                        |   2 +
+ solv/defines.h                             |  28 ++
+ solv/includes.h                            |   7 +-
+ solv/prototypes.h                          |   3 +-
+ solv/tdnfrepo.c                            | 534 ++++++++++++++++++++-
+ tools/cli/lib/help.c                       |   1 +
+ tools/cli/lib/parseargs.c                  |   1 +
+ 13 files changed, 628 insertions(+), 18 deletions(-)
+
+diff --git a/client/config.c b/client/config.c
+index 8ddcc7a..5126e45 100644
+--- a/client/config.c
++++ b/client/config.c
+@@ -85,6 +85,7 @@ TDNFReadConfig(
+     pConf->nInstallOnlyLimit = 1;
+     pConf->nCleanRequirementsOnRemove = 0;
+     pConf->nKeepCache = 0;
++    pConf->nRepoEpochTimeLimit = NULL;
+     pConf->nOpenMax = TDNF_DEFAULT_OPENMAX;
+ 
+     register_ini(NULL);
+@@ -122,6 +123,10 @@ TDNFReadConfig(
+         {
+             pConf->nInstallOnlyLimit = atoi(cn->value);
+         }
++        else if (strcmp(cn->name, TDNF_CONF_KEY_EPOCH_TIME_LIMIT) == 0)
++        {
++            pConf->nRepoEpochTimeLimit = atol(cn->value); //assumes your system's time_t is typedef long
```

Comment:
This should be strol or similar.  atoi is "non-safe"

Reply:
converted the instances of `atoi` added by my code to use `strtoll` instead as suggested.